beetmoverscript/zip.py

```diff
@@ -180,13 +198,43 @@ def _ensure_all_expected_files_are_present_in_archive(zip_path, files_in_archive
                     file_, zip_path
                 )
             )
-        if file_ not in unique_expected_files:
+        if 'SNAPSHOT' in mapping_manifest['s3_bucket_path']:
+            (date, clock, bno) = _extract_and_check_timestamps(file_,
+                                                               SNAPSHOT_TIMESTAMP_REGEX)
+            _ = {
+                'date_timestamp': date,
+                'clock_timestamp': clock,
+                'build_number': bno
+            }
+            # reload the unique_expected_files with their corresponding values
+            # by rendering them via jsone
+            rendered_unique_expected_files = set([jsone.render(f, _) for f in unique_expected_files])
```

Comment:
I'm not totally sold on the usefulness of jsone. If I understand correctly, we're using it because:

* we need to render a file at 2 different times.
* Jinja parsing doesn't error out because some params are missing in the context, during the first parsing.

I think jsone is not really the right tool for this job, because:

* jsone is more complex than just for doing variable replacement. It supports [several operators](https://taskcluster.github.io/json-e/#Language/operators) and has its [own syntax](https://taskcluster.github.io/json-e/#Language/expression-syntax).
* jsone syntax might interfere in oddly ways when first parsed by Jinja. E.g.: what would be the output of:

```jinja
{{ {$eval: 'abs(some_number)'} }}
```
?

If we absolutely need to render a jinja template twice, then what do you think of providing a Jinja context like this one, the first time:

```py
context = {
    'date_timestamp': "{{ date_timestamp }}",    # We still don't know what this value is for now. It will be provided at the second rendering step, once we know what the timestamp actually is. 
}
```
?

Reply:
Sounds good, I addressed that in https://github.com/mozilla-releng/beetmoverscript/pull/194/commits/e444f17ed411b80bd0238a89c72100533dff7f14